src/00b3_survey_mics/script.R

```diff
@@ -211,146 +290,183 @@ mics_final <- circ_recoded %>%
     }
   }) %>%
   # remove empty list elements (handy!) and join data
-  purrr::compact() %>% 
-  bind_rows()
+  purrr::compact() %>%
+  bind_rows() %>%
+  select(-mwm6y, -mwm6m, -mwm6d)
+
+
+#' Soft validation: check <25% missing for expected complete variables
+#' This will ensure that variable not coded entirely as NA (unexpectedly
+#' missed during extract)
+
+check_not_na <- function(var, threshold = 0.25) {
+  mics_final %>%
+    group_by(survey_id) %>%
+    summarise(mean(is.na({{var}})) <= threshold) %>%
+    pull() %>%
+    stopifnot()
+}
+
+check_not_na(dob)
+check_not_na(dob)
+check_not_na(area_name)
+check_not_na(circ_status)
+check_not_na(individual_id, 0.0)
+
+#' Handful of missing area_name for TCD2019MICS, mnweight also =0
+mics_final %>%
+  filter(is.na(area_name))
+
+mics_final <- mics_final %>%
+  filter(! (mnweight == 0 & is.na(area_name) & survey_id == "TCD2019MICS"))
+
+check_not_na(area_name, 0.0)
+
+stopifnot(mics_final$circ_status %in% c(0, 1, NA))
```

Comment:
This validation is a bit _ad hoc_. Enough for here/now, but we should try to formalise this a bit later on. (I think will save some of the data set pain we are having with areas and variable recodings at the modelling stage.)